### ISSUE TITLE
Shadow render failure when no object is present - (crashes some newer android devices!) 

### ISSUE THREAD

**Issue by klauskobald** [wave-harmonic/crest#884]
**Describe the bug**
When "create shadow data" is active on ocean renderer component without any objects producing shadows the whole ocean gets dimmed as if there is a big shadow all over the place except for some places depending on camera angle.
I did not give too much attention to this until one of my Android testers reported to me, that the device is crashing in exact this state during booting. So, it seems as when there is no shadow data produced the shadow lod manager is sending some weird data to the shader.

**Screenshots / video**
[Video](http://clients.kobald.com/dyb/show/crest-shadow-issue.mp4)
When monolith is present, it looks good. When turned off ocean goes darker and blue artefacts appear on the lower screen.

**Versions**
URP 10.5.1
Crest 4.12

**Platform**
 - Editor Mac
 - Android -> crash

**Hardware**
new Huawai, Samsung Galaxy

**Workaround**
I currently turn on shadows when objects are in the scene. That actually can only be done via reflections because the property is private ! Is there a reason for that?

**Comment by klauskobald** [comment]
I discovered more. It seems, when there is no object within the view of the camera this flickering is happening.
The shadowmaps also look somewhat destroyed. 
[movie1](http://clients.kobald.com/misc/crest/crest_shadow_fail_1.mp4)
[movie2](http://clients.kobald.com/misc/crest/crest_shadow_fail_2.mp4)

When camera is close to objects, everything works.
I am using 2 and 4 shadow cascades. Same result.

**Comment by daleeidd** [comment]
Thanks. I'll look into these. I have another report of shadow weirdness.

The first thing I found is that if there is nothing that casts a shadow, the shadow map is unbound. An unbounded texture is usually a grey colour which we treat as shadow. I will look at solving this first.

**Comment by daleeidd** [comment]
If you add the following to _SampleShadowsURP.cs_:

```csharp
if (!renderingData.cullResults.GetShadowCasterBounds(renderingData.lightData.mainLightIndex, out _))
{
    return;
}
```

After:
```csharp
if (renderingData.lightData.mainLightIndex == -1)
    return;
```

It will solve the first issue. Let me know if it also solves the other issues you are seeing too.

**Comment by klauskobald** [comment]
YES! All shadow lods are clean, no more flickering of ocean surface.
I cannot confirm, if this would no longer crash Android but most likely! My suspicion is even that maybe devices are still crashing right now, if a user would turn the camera away from any objects. I think that´s most unlikely to happen, though.
Anyway I would say, that is fixed!

**Comment by daleeidd** [comment]
Thank you. Fix will be out for 4.13.

**Comment by daleeidd** [comment]
4.13 has been released. Thank you for your help.
